pkg/services/govmomi/vcenter/clone_test.go

```diff
@@ -114,32 +121,268 @@ func TestGetDiskSpec(t *testing.T) {
-				primaryDevice := devices[0]
+				primaryDevice := deviceResults[0]
 				validateDiskSpec(t, primaryDevice, tc.cloneDiskSize)
 				if len(tc.additionalCloneDiskSizes) != 0 {
-					secondaryDevice := devices[1]
+					secondaryDevice := deviceResults[1]
 					validateDiskSpec(t, secondaryDevice, tc.additionalCloneDiskSizes[0])
 				}
+
+				// Check number of disks present
+				if len(deviceResults) != tc.expectedDiskCount {
+					t.Fatalf("Expected device count to be %v, but found %v", tc.expectedDiskCount, len(deviceResults))
+				}
+			}
+		})
+	}
+}
+
+func TestCreateDiskSpec(t *testing.T) {
+	model, session, server := initSimulator(t)
+	t.Cleanup(model.Remove)
+	t.Cleanup(server.Close)
+	vm := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+	machine := object.NewVirtualMachine(session.Client.Client, vm.Reference())
+
+	devices, err := machine.Device(ctx.TODO())
+	if err != nil {
+		t.Fatalf("Failed to obtain vm devices: %v", err)
+	}
+	defaultDisks := devices.SelectByType((*types.VirtualDisk)(nil))
+	if len(defaultDisks) < 1 {
+		t.Fatal("Unable to find attached disk for data disk testing")
+	}
+
+	testCases := []struct {
+		expectDevice             bool
+		cloneDiskSize            int32
+		additionalCloneDiskSizes []int32
+		name                     string
+		disks                    object.VirtualDeviceList
+		dataDisks                []infrav1.VSphereDisk
+		expectedDiskCount        int
+		err                      string
+	}{
+		{
+			name:  "Successfully add data disk",
+			disks: devices,
+			dataDisks: []infrav1.VSphereDisk{
+				{
+					SizeGiB: 10,
+				},
+			},
+			expectDevice: true,
+		},
+		{
+			name:  "Successfully add multiple data disks",
+			disks: devices,
+			dataDisks: []infrav1.VSphereDisk{
+				{
+					SizeGiB: 10,
+				},
+				{
+					SizeGiB: 20,
+				},
+			},
+			expectDevice: true,
+		},
+		{
+			name:  "Successfully add multiple data disks when template has multiple disks",
+			disks: append(devices, defaultDisks...),
+			dataDisks: []infrav1.VSphereDisk{
+				{
+					SizeGiB: 10,
+				},
+				{
+					SizeGiB: 20,
+				},
+			},
+			expectDevice: true,
+		},
+	}
+
+	for _, test := range testCases {
+		tc := test
+		t.Run(tc.name, func(t *testing.T) {
+			deviceResults, err := createDataDisks(ctx.TODO(), tc.dataDisks, tc.disks)
+			if (tc.err != "" && err == nil) || (tc.err == "" && err != nil) || (err != nil && tc.err != err.Error()) {
+				t.Fatalf("Expected to get '%v' error from createDataDisks, got: '%v'", tc.err, err)
+			}
+			if deviceFound := len(deviceResults) != 0; tc.expectDevice != deviceFound {
+				t.Fatalf("Expected to get a device: %v, but got: '%#v'", tc.expectDevice, deviceResults)
+			}
+			if tc.expectDevice {
+				// Check number of disks present
+				if len(deviceResults) != len(tc.dataDisks) {
+					t.Fatalf("Expected device count to be %v, but found %v", tc.expectedDiskCount, len(deviceResults))
+				}
+			}
+		})
+	}
+}
+
+func TestAddingDataDisks(t *testing.T) {
```

Comment:
```suggestion
func TestCreateDataDisks_AddingDataDisks(t *testing.T) {
```

Reply:
I merged these tests.  Should be cleaner now.